From c25c0113c8800cdbf632dbe59ac912992db96a92 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Mon, 22 Apr 2019 16:42:05 +0300
Subject: [PATCH 1/1] Completely remove AppMenu.

The Solus GNOME Experience should set forth the expectation that application menus are reliably in their respective applications. GNOME has gone through considerable efforts in 3.32 series to move items into various inner-app menus and therefore we should not be forced to have AppMenu enabled.
---
 js/ui/panel.js         | 7 +------
 js/ui/windowManager.js | 2 +-
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/js/ui/panel.js b/js/ui/panel.js
index 9790d4843..59c21982f 100644
--- a/js/ui/panel.js
+++ b/js/ui/panel.js
@@ -242,10 +242,6 @@ var AppMenuButton = GObject.registerClass({
 
         this._spinner = null;
 
-        let menu = new AppMenu(this);
-        this.setMenu(menu);
-        this._menuManager.addMenu(menu);
-
         let tracker = Shell.WindowTracker.get_default();
         let appSys = Shell.AppSystem.get_default();
         this._focusAppNotifyId =
@@ -813,7 +809,6 @@ class AggregateMenu extends PanelMenu.Button {
 const PANEL_ITEM_IMPLEMENTATIONS = {
     'activities': ActivitiesButton,
     'aggregateMenu': AggregateMenu,
-    'appMenu': AppMenuButton,
     'dateMenu': imports.ui.dateMenu.DateMenuButton,
     'a11y': imports.ui.status.accessibility.ATIndicator,
     'keyboard': imports.ui.status.keyboard.InputSourceIndicator,
@@ -1025,7 +1020,7 @@ class Panel extends St.Widget {
     }
 
     toggleAppMenu() {
-        this._toggleMenu(this.statusArea.appMenu);
+        //this._toggleMenu(this.statusArea.appMenu);
     }
 
     toggleCalendar() {
diff --git a/js/ui/windowManager.js b/js/ui/windowManager.js
index b9f5fef46..73a5a61fb 100644
--- a/js/ui/windowManager.js
+++ b/js/ui/windowManager.js
@@ -2117,7 +2117,7 @@ var WindowManager = class {
     }
 
     _toggleAppMenu(display, window, event, binding) {
-        Main.panel.toggleAppMenu();
+        //Main.panel.toggleAppMenu();
     }
 
     _toggleCalendar(display, window, event, binding) {
-- 
2.21.0

